### PR TITLE
[v9] Don't enforce standard k8s and ssh auth mechanisms when joining sessions

### PR DESCRIPTION
**Pull request by xacrimon**
Backport of:
- #11144 
- #12473

Also includes changes to the tsh join metadata retrieval to special-case sub-v9 servers for compatibility. Only v9+ auths support sessiontracker so we need to use the legacy system to find sessions if the cluster auth is pre-v9.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -1560,54 +1582,47 @@ func (tc *TeleportClient) Join(ctx context.Context, mode types.SessionParticipan
...
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	var session *session.Session
-	for _, s := range sessions {
-		if s.ID == sessionID {
-			session = &s
-			break
+
+	nodeLogin := teleport.SSHSessionJoinPrincipal
```
Comment (espadolini) [inline lib/client/api.go:1591]:
```suggestion
	// TODO(xacrimon): DELETE IN 10.0
	nodeLogin := teleport.SSHSessionJoinPrincipal
```

**Review by espadolini — APPROVED** [review]
LGTM but I'm a bit worried about supporting v8 clusters with a v9 `tsh`; are we ok with user expectations growing like that?

**Comment by xacrimon** [comment]
I'm mostly bolting on support here because it's easy to do, doesn't require serverside changes and because our versioning guide doesn't really specify it + the fact that we've that similar things happen before with users eagerly upgrading. If we decide against it I'd be happy to drop support.

**Review by zmb3 — APPROVED** [review]

**Comment by xacrimon** [comment]
Closing this out since it seems better to not change the permission system in an already released major.

**Pull request closed without merging**